src/Package/Impl/Repl/Commands/ReplCommandController.cs

```diff
@@ -152,10 +152,12 @@ public ReplCommandController(ITextView textView, ITextBuffer textBuffer)
             }
         }
 
-        private void HandleCancel(RCompletionController controller) {
-            Workflow.Operations.CancelAsync().DoNotWait();
-            // Post interrupt command which knows if it can interrupt R or not
-            VsAppShell.Current.PostCommand(RGuidList.RCmdSetGuid, RPackageCommandId.icmdInterruptR);
+        private void HandleCancel(RCompletionController controller, ITextView textView) {
+            if (!controller.HasActiveCompletionSession && !controller.HasActiveSignatureSession(textView)) {
```

Comment:
InterruptR may or may not work. Example is when typing incomplete expression user wants out and hits ESC. It should return user to `>`. However, InterruptR doesnt work for that. Now, if there is a long running process then InterruptR works (and clears pending inputs). Also, InterruptR is disabled when in debugger break mode but ESC should still work.